File: dsp03.pas
(*******  PROGRAM NO.3   AMBIGUITY IN DIGITAL SINUSOIDS  *******)
program DSP03;
uses graph3;
var X: array[1..320] of real;
var N,M,K: integer;
begin
(****  GENERATE SIGNAL WITH TEN EQUAL FREQUENCY INCREMENTS  ****)
  for M:=0 to 9 do
    begin
      for K:=1 to 32 do
        begin
          N:=K+32*M;
          X[N]:=cos(N*2*3.141593*M/8);
        end;
    end;
(**************************  PLOT  *****************************)
  graphmode;
  for N:=1 to 320 do
    begin
      draw(N,50,N,round(50-X[N]*25),1);
    end;
end.
(***************************************************************)



File: dsp03.c
#include <graph.h>
#include <conio.h>
#include <stdio.h>
#include <math.h>

#define XDIM 320

float   fX[XDIM + 1];

void main (void)
    {
    /*
     *****  PROGRAM NO.3   AMBIGUITY IN DIGITAL SINUSOIDS  *****
     */
    int    iCount, iM, iK;

    _clearscreen  (_GCLEARSCREEN);
    /*
     *** GENERATE SIGNAL WITH TEN EQUAL FREQUENCY INCREMENTS ***
     */
    printf ("PROGRAM 3: Ambiguity in digital sinusoids\n");
    printf ("Press any key to start...\n\n");
    getch();

    for (iM = 0; iM < 10; iM++)
        for (iK = 1; iK < 33; iK++)
            {
            iCount = iK + 32 * iM;
            fX[iCount] = cos ((float)iCount * 2.0 *
                         3.141593 * (float)iM / 8.0);
            }

    /*
     *********************  PLOT  ******************************
     */
    _setvideomode( _MRES16COLOR );

    for (iCount = 1; iCount <= XDIM; iCount++)
        {
        _moveto (iCount, 50);
        _lineto (iCount, 50 - (int)(fX[iCount] * 25.0  + 0.5));
        }

    getch();
    _setvideomode( _DEFAULTMODE );

    /*
     ***********************************************************
     */
    }
